doc/modules/mod_http_upload.md

```diff
@@ -7,23 +7,91 @@ Currently, the module supports only the [S3][s3] backend using [AWS Signature Ve
-* **token_bytes** (integer, default: `32`) - Number of random bytes of a token that will be used in a generated URL. 
- The text representation of the token will be twice as long as the number of bytes, e.g. for the default value the token in the URL will be 64 characters long.
-* **max_file_size** (integer, default: 10485760 (10 MB)) - Maximum file size (in bytes) accepted by the module. Disabled if set to `undefined`.
-* **s3** (list, default: unset) - Options specific to [S3][s3] backend.
+#### `modules.mod_http_upload.iqdisc`
+* **Syntax:** string
+* **Default:** `"one_queue"`
+* **Example:** `iqdisc = "one_queue"`
+
+#### `modules.mod_http_upload.host`
+* **Syntax:** string
+* **Default:** `"upload.@HOST@"`
+* **Example:** `host = "upload.@HOST@"`
+
+Subdomain for the upload service to reside under. `@HOST@` is replaced with each served domain.
+
+#### `modules.mod_http_upload.backend`
+* **Syntax:** non-empty string
+* **Default:** `"s3"`
+* **Example:** `backend = "s3"`
+
+Backend to use for generating slots. Currently only `"s3"` can be used.
+
+#### `modules.mod_http_upload.expiration_time`
+* **Syntax:** non-negative integer
+* **Default:** `60`
+* **Example:** `expiration_time = 120`
+
+Duration (in seconds) after which the generated `PUT` URL will become invalid.
+
+#### `modules.mod_http_upload.token_bytes`
+* **Syntax:** positive integer
+* **Default:** `32`
+* **Example:** `token_bytes = 32`
+
+Number of random bytes of a token that will be used in a generated URL. 
+The text representation of the token will be twice as long as the number of bytes, e.g. for the default value the token in the URL will be 64 characters long.
+
+#### `modules.mod_http_upload.max_file_size`
+* **Syntax:** non-negative integer or the string `"undefined"`
+* **Default:** `10485760`
+* **Example:** `max_file_size = "undefined"`
+
+Maximum file size (in bytes) accepted by the module. Disabled if set to `"undefined"`.
+
+#### `modules.mod_http_upload.s3`
+* **Syntax:** Array of TOML tables. See description.
+* **Default:** see description
+* **Example:** see description
+
+Options specific to [S3][s3] backend.
 
 #### [S3][s3] backend options
 
-* **bucket_url** (string, default: unset) - A complete URL pointing at the used bucket. The URL may be in [virtual host form][aws-virtual-host], and for AWS it needs to point to a specific regional endpoint for the bucket. The scheme, port and path specified in the URL will be used to create `PUT` URLs for slots, e.g. specifying a value of `"https://s3-eu-west-1.amazonaws.com/mybucket/custom/prefix"` will result in `PUT` URLs of form `"https://s3-eu-west-1.amazonaws.com/mybucket/custom/prefix/<RANDOM_TOKEN>/<FILENAME>?<AUTHENTICATION_PARAMETERS>"`.
-* **add_acl** (boolean, default: `false`) - If `true`, adds `x-amz-acl: public-read` header to the PUT URL.
+##### `s3.bucket_url`
+* **Syntax:** non-empty string
+* **Default:** `""`
```

Comment:
It was previously described as `default: unset` while it's validated by `validate_non_empty_string/1`. What should be default value here?

Reply:
I would write something like `no default` when there is no default set in the code. This means that using this module without providing this option wouldn't make much sense. Similarly passing an empty string as the value.